andela/mervels-bn-backend
# #168664600 Fix creation of multiple accommodations of the same name in the same location by travel admin

#### What does this PR do?
- it fixes the feature of creating accommodations and adding rooms by travel admin
#### Description of Task to be completed?
`before:`
```
- the travel admin was able to create the accommodations and add rooms to those accommodations
- he could add many accommodations of the same name to the same location
- he could add many rooms of the same name to the same accommodation.
```
`after fix:`
```
- travel admin can create accommodations and add rooms
- no accommodation can have more than one room with the same name
- no location can have more than one accommodation with the same name
```
#### How should this be manually tested?
- clone the repo
- run npm install
- run npm test to run the tests
- run npm run start:dev to start the server
- creating an accommodation and add room
- try adding the same room. 'you should get an error'
- try adding the same accommodation. 'you should get an error'
#### Any background context you want to provide?
**minor changes in the accommodation services**
```
- the method 'getAccomodationById' is changed `getAccommodation` and is no longer accepting the id as a parameter.
- it is receiving an object of key id, and its value eg: {id: 3}
```
#### What are the relevant pivotal tracker stories?
[#168664600](https://www.pivotaltracker.com/story/show/168664600)
#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/39488016/65387901-7bc05a00-dd4b-11e9-8e6e-314cdb58cfd5.png)
![image](https://user-images.githubusercontent.com/39488016/65388000-add1bc00-dd4b-11e9-9952-f036c6c80632.png)
#### Questions:
N/A
